var/spack/repos/builtin/packages/bcache/package.py

```diff
@@ -25,7 +25,9 @@ class Bcache(MakefilePackage):
     depends_on("pkgconfig", type="build")
 
     def setup_build_environment(self, env):
-        env.append_flags("LDFLAGS", "-lintl")
+        # Add -lintl if provided by gettext, otherwise libintl is provided by the system's glibc:
+        if "libintl" in self.spec["gettext"].libs.joined():
```

Comment:
I think `self.spec["gettext"] is self`? Also since `.libs` is essentially a list of strings, it's better to do
```suggestion
        if any("libintl" in filename for filename in self.libs):
```
and hope that the user hasn't installed into a directory named `/my/libintl-doesnt-work/`

Reply:
Whoops I just saw that this is inside the bcache package not gettext package... my bad.